demisto_sdk/commands/pre_commit/pre_commit_command.py

```diff
@@ -151,23 +151,34 @@ def prepare_hooks(
             "all_files": self.all_files,
             "input_mode": self.input_mode,
         }
-        PyclnHook(**hooks["pycln"], **kwargs).prepare_hook(PYTHONPATH)
-        RuffHook(**hooks["ruff"], **kwargs).prepare_hook(
+        PyclnHook(**hooks.pop("pycln"), **kwargs).prepare_hook(PYTHONPATH)
+        RuffHook(**hooks.pop("ruff"), **kwargs).prepare_hook(
             self.python_version_to_files, IS_GITHUB_ACTIONS
         )
-        MypyHook(**hooks["mypy"], **kwargs).prepare_hook(self.python_version_to_files)
-        SourceryHook(**hooks["sourcery"], **kwargs).prepare_hook(
+        MypyHook(**hooks.pop("mypy"), **kwargs).prepare_hook(
+            self.python_version_to_files
+        )
+        SourceryHook(**hooks.pop("sourcery"), **kwargs).prepare_hook(
             self.python_version_to_files, config_file_path=SOURCERY_CONFIG_PATH
         )
-        ValidateFormatHook(**hooks["validate"], **kwargs).prepare_hook(
+        ValidateFormatHook(**hooks.pop("validate"), **kwargs).prepare_hook(
+            self.files_to_run
+        )
+        ValidateFormatHook(**hooks.pop("format"), **kwargs).prepare_hook(
             self.files_to_run
         )
-        ValidateFormatHook(**hooks["format"], **kwargs).prepare_hook(self.files_to_run)
         [
             DockerHook(**hook, **kwargs).prepare_hook(files_to_run=self.files_to_run)
             for hook_id, hook in hooks.items()
             if hook_id.endswith("in-docker")
         ]
+        docker_without_hooks = [
```

Comment:
this will include format and verify hooks and all the others. why only specifically exclude docker?

Reply:
in all of the other hooks (PyclnHook, RuffHook, MypyHook, SourceryHook, ValidateFormatHook) I used `pop()` as we discussed, and I also need to exclude the docker hooks as well before initializing the remaining hooks.